src/test/checkdatasig_tests.cpp

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "test/test_bitcoin.h"
+
+#include "core_io.h"
+#include "utilstrencodings.h"
+#include "policy/policy.h"
+#include "script/interpreter.h"
+#include "unlimited.h"
+
+#include <boost/test/unit_test.hpp>
+
+#include <array>
+
+typedef std::vector<uint8_t> valtype;
+typedef std::vector<valtype> stacktype;
+
+BOOST_FIXTURE_TEST_SUITE(checkdatasig_tests, BasicTestingSetup)
+
+std::array<uint32_t, 3> flagset{
+    {0, STANDARD_SCRIPT_VERIFY_FLAGS, MANDATORY_SCRIPT_VERIFY_FLAGS}};
+
+const uint8_t vchPrivkey[32] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1};
+
+// BU: dummy value.
+const uint32_t SCRIPT_ENABLE_CHECKDATASIG = 1<<31;
+
+struct KeyData {
+    CKey privkey, privkeyC;
+    CPubKey pubkey, pubkeyC, pubkeyH;
+
+    KeyData() {
+        privkey.Set(vchPrivkey, vchPrivkey + 32, false);
+        privkeyC.Set(vchPrivkey, vchPrivkey + 32, true);
+        pubkey = privkey.GetPubKey();
+        pubkeyH = privkey.GetPubKey();
+        pubkeyC = privkeyC.GetPubKey();
+        *const_cast<uint8_t *>(&pubkeyH[0]) = 0x06 | (pubkeyH[64] & 1);
+    }
+};
+
+static std::string StackToString(const stacktype &stack) {
+    std::string res="|";
+    for (const valtype& v : stack) {
+        res+=HexStr(v, true);
+        res+="|";
+    }
+    return res;
+}
+static void CheckError(uint32_t flags, const stacktype &original_stack,
+                       const CScript &script, ScriptError expected) {
+    enableCheckDataSigVerify = (flags & SCRIPT_ENABLE_CHECKDATASIG);
+    BOOST_TEST_INFO("CheckError()");
+    BOOST_TEST_INFO(tfm::format("Flags: %x, enCDS: %d, origStack: %s, Script: %s, Expected Error: %s",
```

Comment:
BOOST_TEST_INFO is not compatible with our current min boost level that we support.

Reply:
Good catch. I guess I can just remove this code for now, it is from debugging.